sharding/collation.go

```diff
@@ -29,9 +30,14 @@ func (c *Collation) Header() *CollationHeader           { return c.header }
 func (c *Collation) Transactions() []*types.Transaction { return c.transactions }
 
 func (c *Collation) SetHeader(h *CollationHeader) { c.header = h }
-func (c *Collation) AddTransaction(tx *types.Transaction) {
+func (c *Collation) AddTransaction(tx *types.Transaction) error {
 	// TODO: Check transaction does not exceed gas limit
+	gasLimit := uint64(4000000)
```

Comment:
where did we get 4000000 from?

Reply:
It is an arbitrary number , I have not found any information specifying a gas limit for a transaction in a collation so I just used something close to the block gas limit on the main chain

Reply:
I see. I was just wondering because I couldn't find any information on gas limit either
Block gas limit is ~8000000 according to https://ethstats.net/

Reply:
Yeah the gas limit keeps on increasing lol , so I dont know what would be a good number to use.

Reply:
Agree with @prestonvanloon. Not a relevant consideration until we understand the proposal system more.